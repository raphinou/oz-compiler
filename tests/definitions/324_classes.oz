% Dynamic attribute access
local
   class Accessors
      meth set(Attr Value)
         Attr:=Value
      end
      meth get(Attr ?R)
         R=@Attr
      end
   end
   class Symbol from Accessors
      attr
        id
         name
         pos
      meth init(Id Name Pos)
         id:=Id
         name:=Name
         pos:=Pos
      end
   end
   I={New Symbol init(1 'S1' pos(1 2))}
in
   {Show {I get(pos $)}}
   {Show {I get(name $)}}
   {I set(name 'S2')}
   {Show {I get(name $)}}
end

