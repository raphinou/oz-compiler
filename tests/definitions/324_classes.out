pos(1 2)
'S1'
'S2'
